ipassign/protocol/message.py

```diff
@@ -108,24 +101,26 @@ def payload(self):
 
     @payload.setter
     def payload(self, val):
-        if isinstance(val, Payload):
+        if isinstance(val, (Acknowledgement, Configuration)):
             pass
         elif isinstance(val, bytes):
-            if len(val) != 0:
-                val = Payload.from_bytes(val)
+            if len(val) == 56:
+                val = Configuration.from_bytes(val)
+            if len(val) == 4:
+                val = Acknowledgement.from_bytes(val)
```

Comment:
This should be an `if/elif/else`:
```python
if len(val) == 56:
    val = Configuration.from_bytes(val)
elif len(val) == 4:
    val == Acknowledgement.from_bytes(val)
else:
    raise ValueError(f'Payload length should be 4 or 56, not {len(val)}'
```
With the appropriate unit test.

Reply:
Actually, an empty byte array should also be legal, as the discovery message does not have a payload.